plutus-pab/src/Cardano/Wallet/Server.hs

```diff
@@ -98,25 +98,24 @@ app nodeClientEnv chainIndexEnv mVarState =
     (submitTxn >=> const (pure NoContent)) :<|> ownPubKey :<|> uncurry updatePaymentWithChange :<|>
     walletSlot :<|> ownOutputs
 
-main :: MonadIO m => Config -> BaseUrl -> BaseUrl -> Availability -> m ()
-main Config {baseUrl, wallet} nodeBaseUrl chainIndexBaseUrl availability = runStdoutLoggingT $ do
-    let port = baseUrlPort baseUrl
-        state = emptyWalletState wallet
-    nodeClientEnv <-
-        liftIO $ do
-            nodeManager <- newManager defaultManagerSettings
-            pure $ mkClientEnv nodeManager nodeBaseUrl
-    chainIndexEnv <-
-        liftIO $ do
-            chainIndexManager <- newManager defaultManagerSettings
-            pure $ mkClientEnv chainIndexManager chainIndexBaseUrl
+main :: Trace IO WalletMsg -> Config -> NodeUrl -> ChainIndexUrl -> Availability -> IO ()
+main trace Config {..} nodeBaseUrl chainIndexBaseUrl availability = runLogEffects trace $ do
+    nodeClientEnv <- buildEnv nodeBaseUrl defaultManagerSettings
+    chainIndexEnv <- buildEnv chainIndexBaseUrl defaultManagerSettings
     mVarState <- liftIO $ newMVar state
-    _ <- liftIO
-            $ runM
-            $ flip handleError (error . show @ClientError)
-            $ ChainIndexClient.handleChainIndexClient chainIndexEnv
-            $ startWatching (Wallet.ownAddress state)
-    let warpSettings :: Warp.Settings
-        warpSettings = Warp.defaultSettings & Warp.setPort port & Warp.setBeforeMainLoop (available availability)
-    logInfoN $ "Starting wallet server on port: " <> tshow port
+    runClient chainIndexEnv
+    logInfo $ StartingWallet servicePort
     liftIO $ Warp.runSettings warpSettings $ app nodeClientEnv chainIndexEnv mVarState
+    where
```

Comment:
So one style thing that I think we *do* usually go for is to prefer define-before-use, i.e. `let` over `where` unless it's a really big thing.

I am a bit atypical in my anti-where prejudices, though :)